cpp/src/parquet/encryption/internal_file_decryptor.cc

```diff
@@ -215,4 +217,53 @@ std::shared_ptr<Decryptor> InternalFileDecryptor::GetColumnDecryptor(
   return column_data_map_[column_path];
 }
 
+namespace {
+
+std::shared_ptr<Decryptor> GetColumnDecryptor(
+    const ColumnCryptoMetaData* crypto_metadata, InternalFileDecryptor* file_decryptor,
+    const std::function<std::shared_ptr<Decryptor>(
+        InternalFileDecryptor* file_decryptor, const std::string& column_path,
+        const std::string& column_key_metadata, const std::string& aad)>& func) {
+  if (crypto_metadata == nullptr) {
+    return nullptr;
+  }
+
+  if (file_decryptor == nullptr) {
+    throw ParquetException("RowGroup is noted as encrypted but no file decryptor");
+  }
+
+  if (crypto_metadata->encrypted_with_footer_key()) {
+    return file_decryptor->GetFooterDecryptorForColumnMeta();
```

Comment:
Hmm... it should be `GetFooterDecryptorForColumnData` if called from `GetColumnDataDecryptor`, shouldn't it?

Reply:
I'm curious why this doesn't fail the tests, though.

Reply:
Ah, this might be because the only distinction between data and metadata is when selecting the actual cipher, and only if the selected algorithm id is not already `AES_GCM_V1`. Here is for example `AesDecryptor`:
```c++
AesDecryptor::AesDecryptorImpl::AesDecryptorImpl(ParquetCipher::type alg_id, int key_len,
                                                 bool metadata, bool contains_length) {
  ctx_ = nullptr;
  length_buffer_length_ = contains_length ? kBufferSizeLength : 0;
  ciphertext_size_delta_ = length_buffer_length_ + kNonceLength;
  if (metadata || (ParquetCipher::AES_GCM_V1 == alg_id)) {
    aes_mode_ = kGcmMode;
    ciphertext_size_delta_ += kGcmTagLength;
  } else {
    aes_mode_ = kCtrMode;
  }
```

Reply:
Good catch! Sorry for the mistake made by the copy-and-paste.